source/common/network/cidr_range.cc

```diff
@@ -0,0 +1,186 @@
+  } else {
+    return other.version() == IpVersion::v6 && ipv6()->address() == other.ipv6()->address();
+  }
+}
+
+const Ipv4* CidrRange::ipv4() const {
+  if (address_ != nullptr) {
+    return address_->ip()->ipv4();
+  }
+  return nullptr;
+}
+
+const Ipv6* CidrRange::ipv6() const {
+  if (address_ != nullptr) {
+    return address_->ip()->ipv6();
+  }
+  return nullptr;
+}
+
+int CidrRange::length() const { return length_; }
+
+IpVersion CidrRange::version() const {
+  if (address_ != nullptr) {
+    return address_->ip()->version();
+  }
+  return IpVersion::v4;
+}
+
+bool CidrRange::isInRange(InstancePtr address) const {
+  if (address == nullptr || length_ < 0 || address->type() != Type::Ip ||
+      address->ip()->version() != version()) {
+    return false;
+  }
+  // Make an CidrRange from the address, of the same length as this. If the two ranges have
+  // are the same, then the address is in this range.
+  CidrRange other = create(address, length_);
+  ASSERT(length() == other.length());
+  ASSERT(version() == other.version());
+  return *this == other;
+}
+
+std::string CidrRange::asString() const {
+  if (address_ == nullptr) {
+    return "/-1";
+  } else {
+    return fmt::format("{}/{}", address_->ip()->addressAsString(), length_);
+  }
+}
+
+// static
+CidrRange CidrRange::create(InstancePtr address, int length) {
+  InstancePtr ptr = truncateIpAddressAndLength(std::move(address), &length);
+  return CidrRange(std::move(ptr), length);
+}
+
+// static
+CidrRange CidrRange::create(const std::string& address, int length) {
+  return create(parseInternetAddress(address), length);
+}
+
+// static
+CidrRange CidrRange::create(const std::string& range) {
+  std::vector<std::string> parts = StringUtil::split(range, '/');
+  if (parts.size() == 2) {
+    InstancePtr ptr = parseInternetAddress(parts[0]);
+    if (ptr != nullptr && ptr->type() == Type::Ip) {
+      uint64_t length64;
+      if (StringUtil::atoul(parts[1].c_str(), length64, 10)) {
+        if ((ptr->ip()->version() == IpVersion::v6 && length64 <= 128) ||
+            (ptr->ip()->version() == IpVersion::v4 && length64 <= 32)) {
+          return create(std::move(ptr), static_cast<uint32_t>(length64));
+        }
+      }
+    }
+  }
+  return CidrRange(nullptr, -1);
+}
+
+// static
+InstancePtr CidrRange::truncateIpAddressAndLength(InstancePtr address, int* length_io) {
+  int length = *length_io;
+  if (address == nullptr || length < 0 || address->type() != Type::Ip) {
+    *length_io = -1;
+    return nullptr;
+  }
+  switch (address->ip()->version()) {
+  case IpVersion::v4: {
+    if (length >= 32) {
+      // We're using all of the bits, so don't need to create a new address instance.
+      *length_io = 32;
+      return address;
+    } else if (length == 0) {
+      // Create an Ipv4Instance with only a port, which will thus have the any address.
+      // TODO( Consider adding a Utility::AnyIpv4Address() static method to return such an instance.
```

Comment:
Done.
From that leading paren, looks like I started to think about whether I was supposed to include a name or not, but then didn't remember to come back to it.
AND, I have no removed that TODO because I added such a function to Utility.